tests/scripts/all.sh

```diff
@@ -994,7 +1000,7 @@ component_check_doxygen_warnings () {
 
 component_test_default_out_of_box () {
     msg "build: make, default config (out-of-box)" # ~1min
-    make
+    make CFLAGS="$GLOBAL_CFLAGS"
```

Comment:
When you pass `CFLAGS` explicitly, it overrides the default `CFLAGS` in the makefile. That's
```
CFLAGS  ?= -O2
```
present in `library/Makefile`, `programs/Makefile` and `tests/Makefile`. So now all the builds for which `all.sh` doesn't have an explicit `-O` flag will run with no `-O` flag, i.e. no optimization. This is bad for several reasons: slower to run tests, misses some compiler warnings, not testing the impact of compiler optimizations.

I haven't checked whether builds using cmake have a similar issue.

I am also concerned that we'll forget to set `CFLAGS` in some builds, or get it wrong. (Historical precedent: we used to forget to pass a `-O` flag when amending `CFLAGS` for Asan, which is why we introduced `ASAN_CFLAGS`.) So in practice, over time, we'll end up with some components not using the new bignum, and some component not doing optimized builds.

Please use a different approach that does not require changing most `all.sh` components. For example, it would be enough to add a setting to the make and cmake files.

Reply:
The CMake version does not appear to overwrite default CFlags. I tested this previously, as I had similar concerns.